src/genie/libs/parser/nxos/show_eigrp.py

```diff
@@ -386,3 +386,161 @@ def cli(self, vrf='all', output=None):
+                        '(,\s(?P<interface>\S+))?$')
+
+        parsed_dict:dict = {}
+
+        nexthop_count:int = 0
+
+        # loop over all the lines
+        for line in output.splitlines():
+            line = line.strip()
+
+            result = r1.match(line)
+            if result:
+                group:dict = result.groupdict()
+                address_family = group['address_family'].lower()
+                if address_family == 'ip':
+                    address_family = 'ipv4'
+                as_num = group['as_num']
+                routerid = group['routerid']
+                vrf = group['vrf']
+                as_dict:dict = parsed_dict \
+                    .setdefault('as', {}) \
+                    .setdefault(as_num, {})
+                as_dict['routerid'] = routerid
+                continue
+
+            result = r2.match(line)
+            if result:
+                nexthop_count = 0
+                group:dict = result.groupdict()
+
+                if not vrf:
+                    vrf = 'default'
+                if not as_num:
+                    as_num = 0
+                if not routerid:
+                    routerid = ''
+
+                route:str = group['route']
+                route_dict:dict = parsed_dict \
+                    .setdefault('as', {}) \
+                    .setdefault(as_num, {}) \
+                    .setdefault('vrf', {}) \
+                    .setdefault(vrf, {}) \
+                    .setdefault('address_family', {}) \
+                    .setdefault(address_family, {}) \
+                    .setdefault('route', {}) \
+                    .setdefault(route, {})
+                route_dict['state'] = group['state']
+                route_dict['successors'] = int(group['successors'])
+                route_dict['fd'] = int(group['fd'])
+
+            result = r3.match(line)
+            if result:
+                group:dict = result.groupdict()
+                nexthop_dict:dict = parsed_dict \
+                    .setdefault('as', {}) \
+                    .setdefault(as_num, {}) \
+                    .setdefault('vrf', {}) \
+                    .setdefault(vrf, {}) \
+                    .setdefault('address_family', {}) \
+                    .setdefault(address_family, {}) \
+                    .setdefault('route', {}) \
+                    .setdefault(route, {}) \
+                    .setdefault('nexthops', {}) \
+                    .setdefault(nexthop_count, {})
+                nexthop_count = nexthop_count + 1
+                nexthop_dict['nexthop'] = group['nexthop']
+                for key in ('fd', 'rd') :
+                    if group[key]:
+                        nexthop_dict[key] = int(group[key])
+                if group['interface']:
+                    nexthop_dict['interface'] = group['interface']
+
+        return parsed_dict
+
+class ShowIpv4EigrpTopology(ShowEigrpTopologySuperParser,
```

Comment:
no need to create an inheritance for multiple show commands
for example: https://github.com/CiscoTestAutomation/genieparser/blob/master/src/genie/libs/parser/iosxe/show_arp.py#L73